app/controllers/account/tasks_controller.rb

```diff
@@ -36,12 +38,24 @@ def update
 
   def destroy
     resource.destroy
-    redirect_to account_workspace_project_path(parent.workspace_id, parent.id)
+    redirect_to account_workspace_project_path(parent.workspace_id, parent)
   end
 
   def move
-    resource.update_attributes(task_movement_params)
-    redirect_to account_workspace_project_path(parent.workspace_id, parent.id)
+    resource.update(task_movement_params)
+    redirect_to account_workspace_project_path(parent.workspace_id, parent)
+  end
+
+  def watch
+    @task = resource
+
+    if @task.users.include? current_user
+      @task.watches.find_by(user_id: current_user).destroy
+    else
+      @task.update(users: [current_user])
```

Comment:
this is not good. If you have 5 followers already - this code will delete all others I guess ... check it

Reply:
I expect code something like

@task.add_watcher(current_user)
@task.remove_watcher(current_user)